testcases/kernel/syscalls/chmod/chmod08.c

```diff
@@ -0,0 +1,64 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later
+ * Copyright (C) 2020 Invisible Things Lab
+ *                    Michał Kowalczyk <mkow@invisiblethingslab.com>
+ */
+
+/*
+ * DESCRIPTION
+ *  Changes file access permissions using `chmod` with bits outside of 07777 in
+ *  `mode` set and verifies if they were ignored.
+ *
+ * WARNING
+ *  The fact that these bits are ignored is not documented (at the time of
+ *  writing). Failure of this test doesn't necessarily mean that a regression
+ *  in Linux was introduced, its intention is to catch accidental interface
+ *  changes and warn kernel developers if that happens.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "tst_test.h"
+
+#define OPEN_MODE  0644
+#define CHMOD_MODE (0777 | ~07777)
+#define TESTFILE   "testfile"
+
+void test_chmod(void)
+{
+	struct stat stat_buf;
+
+	TEST(chmod(TESTFILE, CHMOD_MODE));
+	if (TST_RET == -1) {
+		tst_res(TFAIL, "chmod(%s, %#o) failed", TESTFILE, CHMOD_MODE);
+	}
+
+	if (stat(TESTFILE, &stat_buf) == -1) {
+		tst_brk(TFAIL | TTERRNO, "stat failed");
+	}
+
```

Comment:
We do have a safe macros to simplify error handling. So this if () could be replaced and is equivalent to just:

``
SAFE_STAT(TESTFILE, &stat_buf);
``